### DISCUSSION TITLE
Is there a Next-auth Google Login new popup window option

### DISCUSSION THREAD

**Discussion by elmcapp** [nextauthjs/next-auth#8267] [Help]
I have a Nextjs project. I am using Next-Auth with Google login. Is there an option when we press button to sign in to get a new pop up window and sign in. Current when user press the sign in button it redirects them to Google auth sign in page. Once they finish with sign in it will take them back to my site. I was hoping to have a new window where they can site in. I don't want my users leaving my site. 

````
"use client";
import React from "react";
import { signIn, signOut, useSession } from "next-auth/react";

function SigninBtn() {
  const { data: session } = useSession();

  console.log(session);

  if (session && session.user) {
    return (
      <div className="flex gap-4 ml-auto">
        <p className="text-sky-600">{session.user.name}</p>
        <button onClick={() => signOut()} className="text-red-600">
          Sign Out
        </button>
      </div>
    );
  }

  return (
    <button
      onClick={() => signIn("google", { redirect: false })}
      className="text-green-600 ml-auto"
    >
      Sign In
    </button>
  );
}

export default SigninBtn;

````

**Comment by future-leader1** [comment]
hey there , you can totally make it so that when users click the sign-in button, a nice little pop-up window shows up for them to sign in through Google. This way, they won't have to leave your site. try below



function SigninBtn() {
  const { data: session } = useSession();

  console.log(session);

  if (session && session.user) {
    return (
      <div className="flex gap-4 ml-auto">
        <p className="text-sky-600">{session.user.name}</p>
        <button onClick={() => signOut()} className="text-red-600">
          Sign Out
        </button>
      </div>
    );
  }

  return (
    <button
      onClick={() => {
        const authWindow = window.open("", "GoogleAuthWindow", "width=800,height=600");
        signIn("google", { callbackUrl: "false" }).then((response) => {
          if (response.error) {
            authWindow.close();
          } else {
            authWindow.location = response.url;
          }
        });
      }}
      className="text-green-600 ml-auto"
    >
      Sign In
    </button>
  );
}



This bit of code works like a charm. When your users hit that sign-in button, a cute little popup window appears. They can sign in through Google right there, all cozy in that window. And the best part? Your site stays put – no wandering off to another page. So, happy signing in, without the travel plans! 🚀🔐

**Comment by arye321** [comment]
https://github.com/arye321/nextauth-google-popup-login

**Comment by eduardojmrosa** [comment]
i tried every approach here but none of it worked out, when i call the signin function even with the redirect false it does the redirect and dont return the URL for oAuth with google POPUP, what can i do ?
